mayastor/src/subsys/config/pool.rs

```diff
@@ -0,0 +1,181 @@
+use std::{fmt::Display, fs, path::Path};
+
+use futures::lock::Mutex;
+use once_cell::sync::{Lazy, OnceCell};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    bdev::VerboseError,
+    core::{Cores, Reactor, Share},
+    lvs::Lvs,
+    pool::PoolsIter,
+    replica::{ReplicaIter, ShareType},
+};
+
+static CONFIG_FILE: OnceCell<String> = OnceCell::new();
+
+/// Initialise the config file location
+fn init_config_location<P>(file: P)
+where
+    P: AsRef<Path> + Display + ToString,
+{
+    CONFIG_FILE.get_or_init(|| file.to_string());
+}
+
+#[inline]
+/// Return the config file location
+fn get_config_location() -> Option<&'static String> {
+    CONFIG_FILE.get()
+}
+
+#[derive(Debug, Default, PartialEq, Serialize, Deserialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct PoolConfig {
+    pub pools: Option<Vec<Pool>>,
+}
+
+impl PoolConfig {
+    /// Load pool configuration from a file
+    pub fn load<P>(file: P) -> Result<PoolConfig, serde_yaml::Error>
+    where
+        P: AsRef<Path> + Display + ToString,
+    {
+        init_config_location(&file);
+
+        let bytes = fs::read(&file).unwrap_or_default();
+
+        if bytes.is_empty() {
+            return Ok(PoolConfig::default());
+        }
+
+        serde_yaml::from_slice(&bytes)
+    }
+
+    /// Write this pool configuration to a file
+    fn write<P>(&self, file: P) -> Result<(), std::io::Error>
+    where
+        P: AsRef<Path>,
+    {
+        let config = serde_yaml::to_string(&self).map_err(|error| {
+            std::io::Error::new(
+                std::io::ErrorKind::Other,
+                format!("pool configuration serialization error: {}", error),
+            )
+        })?;
+
+        fs::write(&file, config.as_bytes())
+    }
+
+    /// Export current pool configuration
+    pub async fn export() -> Result<(), std::io::Error> {
+        static MUTEX: Lazy<Mutex<u32>> = Lazy::new(|| Mutex::new(0));
+
+        if let Some(file) = get_config_location() {
+            info!("Saving pool configuration to {}", file);
+            let _guard = MUTEX.lock().await;
+            return PoolConfig::get().write(file);
+        }
+
+        Ok(())
+    }
+
+    /// Create all pools specified in this pool configuration
+    async fn create_pools(&self) -> usize {
+        let mut failures = 0;
+        if let Some(pools) = self.pools.as_ref() {
+            for pool in pools {
+                info!("creating pool {}", pool.name);
+                if let Err(error) = Lvs::create_or_import(pool.into()).await {
+                    error!(
+                        "Failed to create pool {}: {}",
+                        pool.name,
+                        error.verbose()
+                    );
+                    failures += 1;
+                }
+            }
+        }
+        failures
+    }
+
+    /// Import pools
```

Comment:
We do not need to run this in the context of mayastor, Read the file, and then for each file submit an RPC request